test/files/neg/t8685.scala

```diff
@@ -0,0 +1,29 @@
+
+
+@deprecated("class C is depr", since="now")
+case class C(i: Int)
+
+case class D @deprecated("ctor D is depr", since="now") (i: Int)
+
+@deprecated("class E is depr", since="now")
+case class E(i: Int)
+@deprecated("module E is depr", since="now")
+object E
+
+@deprecated("module F is depr", since="now")
+object F {
+  case class G(i: Int)
+}
+
+trait Applies {
+  def f = C(42)
+  def g = D(42)
+  def h = E(42)
+  def i = F.G(42)
```

Comment:
BTW, the current code does have calls to `checkUndesired` with the match on `RefTree`, so I could have pushed the ball over the line. hashtag americanfootball